File: php/get_endpoints_by_type.php
<?php

// Information obtained from authentication

$userId = '';
$token = '';
$type = ''; // types can be acquired by calling "Get Endpoints"
$url = 'https://api.smartproxy.com/v1/endpoints/' . $type;

$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "GET");

$headers = array();
$headers[] = "Accept: application/json";
$headers[] = "Authorization: Token $token";
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

$result = curl_exec($ch);
if (curl_errno($ch)) {
    echo 'Error:' . curl_error($ch);
}
curl_close($ch);
echo $result;

File: csharp/get_endpoints_by_type.cs
using System;
using System.IO;
using System.Net;

namespace Smartproxy_API
{
    class get_endpoints_by_type
    {
        static void Main(string[] args)
        {
            // Details from authentication
            string token = "";
            string type = ""; // sticky/random available, also can be obtained from get_endpoints.cs

            string uri = "https://api.smartproxy.com/v1/endpoints/" + type;

            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
            request.AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
            request.PreAuthenticate = true;
            request.Headers.Add("Authorization", "Token " + token);
            request.ContentType = "application/json";
            request.Method = "GET";

            using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
            using (Stream stream = response.GetResponseStream())
            using (StreamReader reader = new StreamReader(stream))
            {
                Console.WriteLine(reader.ReadToEnd());
                Console.ReadKey();
            }
        }
    }
}
